### PR TITLE
poll after previous request completes instead of on an interval

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/download_async_modal.js [hunk 1]
```diff
@@ -19,31 +19,37 @@ $(function(){
...
+                        success: function (resp) {
+                            if (resp.trim().length) {
+                                self.$downloading.hide();
+                                self.$download_progress.show().html(resp);
+                                if (self.$download_progress.find(".alert-success").length) {
+                                    keep_polling = false;
+                                }
                             }
+                            if (keep_polling) {
+                                setTimeout(pollDownloadStatus, self.POLL_FREQUENCY)
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/download_async_modal.js]: Missing semicolon.

File: corehq/apps/reports/static/reports/javascripts/reports.download_export.js [hunk 2]
```diff
@@ -15,23 +15,26 @@ var HQExportDownloader = function (options) {
                 $modal.find('.modal-header h3 span').text($(this).data("formname"));
 
                 $.getJSON($(this).data('dlocation'), function (d) {
-                    var autoRefresh = '';
+                    var autoRefresh = true;
                     var pollDownloader = function () {
-                        if ($('#ready_'+d.download_id).length == 0)
+                        if (autoRefresh && $('#ready_'+d.download_id).length == 0)
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/reports.download_export.js]: Use '===' to compare with '0'.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 3]
```diff
@@ -46,9 +46,9 @@ var ExportManager = function (o) {
          * @param params
          */
         updateModal = function(params) {
-            var autoRefresh = '';
+            var autoRefresh = true;
             var pollDownloader = function () {
-                if ($('#ready_'+params.data.download_id).length == 0) {
+                if (autoRefresh && $('#ready_'+params.data.download_id).length == 0) {
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Use '===' to compare with '0'.

**Comment by dannyroberts** [comment]
@snopoke nice change, curious what prompted you to think of it!

**Comment by snopoke** [comment]
I was downloading something in prod and noticed a bunch of requests queue
up together which seemed silly.

**Comment by czue** [comment]
nice. looks good - i think good to merge if you're comfortable not going through QA

**Pull request merged**